spec/mongo/collection_spec.rb

```diff
@@ -2000,6 +2000,41 @@ def generate
   end
 
   describe '#count_documents' do
+
+    context 'no argument provided' do
+
+    # populate documents to add to collection
+      let(:documents) do
+        documents = []
+        1.upto(10) do |index|
+          documents << { key: 'a', _id: "in#{index}" }
+        end
+        documents
+      end
```

Comment:
Nice! This is way cleaner. Now I would go back to the organization you had before:

```
context 'no argument provided' do
   context 'when collection is empty' do
      ...
   end

   context 'when collection is not empty' do
      ...
   end
end
```

That simplifies your setup (only need to keep track of one collection) and makes the test output easier to read.